payjoin/src/core/persist.rs

```diff
@@ -512,12 +523,14 @@ trait InternalSessionPersister: SessionPersister {
         Err: std::error::Error,
     {
         let RejectFatal(event, error) = reject_fatal;
+        let should_close = event.is_closed_event();
         if let Err(e) = self.save_event(event) {
             return InternalPersistedError::Storage(e);
         }
-        // Session is in a terminal state, close it
-        if let Err(e) = self.close() {
-            return InternalPersistedError::Storage(e);
+        if should_close {
```

Comment:
Stylistic nit: unnecessary assignment

```suggestion
        if event.is_closed_event() {
```